### PR TITLE
HPCC-15656: Add the ability to log Response Time

### PR DESCRIPTION
**Pull request by gaojx**
- add ResponseTime in second to logging

Signed-off-by: Jiafu Gao Jiafu.Gao@lexisnexis.com

### CODE REVIEW

**Comment by gaojx** [comment]
@wangkx please code review

**Review by wangkx — APPROVED** [review]
@gaojx looks fine.

**Comment by gaojx** [comment]
@richardkchapman this PR is ready to merge.

**Review by richardkchapman — CHANGES_REQUESTED** [review]
THe commit message says it is logging the response time in ms, but this code looks to me as though it is converting to seconds (/1000).

I would have thought ms was preferable, in which case shouldn't the /1000 be removed?

**Comment by richardkchapman** [comment]
@gaojx One question

**Comment by gaojx** [comment]
The ms is a typo. It should be in seconds. In all our transaction log, the response_time is decimal(8,4), and the unit is second.

**Comment by richardkchapman** [comment]
What you're storing will be truncated to the complete second... if you want a decimal(8,4) then you should do a setPropString and a sprintf, or something like that, I would have thought?

**Comment by gaojx** [comment]
You are right. Will make the change.

**Comment by gaojx** [comment]
@wangkx please review.

File: esp/logging/loggingmanager/loggingmanager.cpp [hunk 1]
```diff
@@ -123,7 +123,8 @@ bool CLoggingManager::updateLog(const char* option, IEspContext& espContext, IPr
         const char* userId = espContext.queryUserId();
         if (userId && *userId)
             espContextTree->addProp("UserName", userId);
-        espContextTree->addPropInt64("ResponseTime", (__int64)(msTick()-espContext.queryCreationTime())/1000);
+
+        espContextTree->addProp("ResponseTime", VStringBuffer("%.4f", (msTick()-espContext.queryCreationTime())/1000.0));
```
Comment (wangkx) [inline esp/logging/loggingmanager/loggingmanager.cpp]: You may need to add .str() after the VStringBuffer to match: void addProp(const char *xpath, const char *val).

Comment (gaojx) [inline esp/logging/loggingmanager/loggingmanager.cpp]: I think it is optional: the operator const char* () will do this conversation automatically.

**Pull request merged**